web/src/engine/keyboard/src/keyboards/loaders/nodeKeyboardLoader.ts

```diff
@@ -22,15 +25,20 @@ export class NodeKeyboardLoader extends KeyboardLoaderBase {
     }
   }
 
-  protected loadKeyboardInternal(uri: string, errorBuilder: KeyboardLoadErrorBuilder): Promise<Keyboard> {
+  protected async loadKeyboardBlob(uri: string): Promise<Blob> {
     // `fs` does not like 'file:///'; it IS "File System" oriented, after all, and wants a path, not a URI.
-    if(uri.indexOf('file:///') == 0) {
+    if (uri.indexOf('file:///') == 0) {
       uri = uri.substring('file:///'.length);
     }
 
+    const buffer = await readFile(uri);
+    return new Blob([buffer]);
+  }
+
+  protected async loadKeyboardFromScript(scriptSrc: string, errorBuilder: KeyboardLoadErrorBuilder): Promise<Keyboard> {
     let script;
     try {
-      script = new vm.Script(fs.readFileSync(uri).toString());
+      script = new vm.Script(scriptSrc);
     } catch (err) {
       return Promise.reject(errorBuilder.missingError(err));
```

Comment:
This should be scriptError surely? And given we are using `async`/`await` here now, we can probably:

```suggestion
      throw errorBuilder.missingError(err);
```